crates/fs/private/hamt/hash.rs

```diff
@@ -0,0 +1,100 @@
+// TODO(appcypher): Based on ipld_hamt implementation
+
+use anyhow::Result;
+use sha3::{Digest, Sha3_256};
+
+use crate::error;
+
+use super::error::HamtError;
+
+//--------------------------------------------------------------------------------------------------
+// Constants
+//--------------------------------------------------------------------------------------------------
+
+const HASH_BYTES: usize = 32;
+const MAX_CURSOR_DEPTH: usize = HASH_BYTES * 2;
+
+//--------------------------------------------------------------------------------------------------
+// Type Definition
+//--------------------------------------------------------------------------------------------------
+
+pub type HashOutput = [u8; HASH_BYTES];
+
+pub trait GenerateHash {
+    fn generate_hash<K: AsRef<[u8]>>(key: &K) -> HashOutput; // &[u8]
+}
+
+pub struct HashQuartets<'a> {
```

Comment:
Googling around, it looks like quartet was used for 4-bit segments in telecommunications stuff. I didn't know that term before!
Maybe it's just me, but there's some more common terms like "Nibble" "half-byte" or "octet". Would they work better?

Reply:
Let's do nibble